Pj/replace/react fe

pablisch opened the pull request:
Added the new React FE. I left the old one in case there are any issues at all and it can be deleted at any point.

I moved a couple of things around in the BE and made a script to watch the BE TS, and start BE and FE.

I moved the app start into server.js as creating the app and running it in the same file can cause test issues. I never work with index.ts/js and now I am thinking having index and server is a bit weird. I should have moved the app in app.ts and left index. Not to bothered but we should talk about this. Does not affect this PR really.

siqbal181 commented:
@pablisch Can we pop the dist files in the gitignore please :) all good otherwise

pablisch commented:
> @pablisch Can we pop the dist files in the gitignore please :) all good otherwise

Well that was harder than it needed to be. dist just would not go! Thank goodness for StackOverflow as I dread to think what nonsense chatGPT would have come up with.

I renamed index and server to app and index which makes more sense. There was a dev script which I am not sure what it did as it targeted the TS files, I ran it and it returned an error so I deleted it as it seemed like it might be trying to do what the server script does anyway.

As soon as we are happy that this FE works (it is the same as in Aura but for TS) then we can delete the old CRA FE.
